pkg/installer/directcsi.go

```diff
@@ -0,0 +1,98 @@
+// This file is part of MinIO Direct CSI
+// Copyright (c) 2021 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package installer
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/minio/direct-csi/pkg/utils"
+)
+
+func getInstaller() (Installer, error) {
+	versionInfo, err := utils.GetDiscoveryClient().ServerVersion()
+	if err != nil {
+		return nil, err
+	}
+	major, err := strconv.Atoi(versionInfo.Major)
+	if err != nil {
+		return nil, fmt.Errorf("Invalid major version: %s", versionInfo.Major)
+	}
+	minor, err := strconv.Atoi(versionInfo.Minor)
+	if err != nil {
+		return nil, fmt.Errorf("Invalid minor version: %s", versionInfo.Minor)
+	}
+
+	switch {
+	case major == 1 && minor == 18:
+		return &v1dot18{}, nil
+	case major == 1 && minor == 19:
+		return &v1dot19{}, nil
+	case major == 1 && minor == 20:
+		return &v1dot20{}, nil
+	case major == 1 && minor == 21:
+		return &v1dot21{}, nil
+	case major == 1 && minor == 22:
+		return &v1dot22{}, nil
+	default:
+		return nil, fmt.Errorf("direct-csi is not supported in kubernetes version: %s.%s", versionInfo.Major, versionInfo.Minor)
+	}
+}
+
+func validateConfig(config *InstallConfig) error {
```

Comment:
Have `validate()` method for `Config` and use this method on first use of config.